sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -658,6 +720,17 @@ class Analyzer(
           Generate(newG.asInstanceOf[Generator], join, outer, qualifier, output, child)
         }
 
+      // A special case for View, replace the output attributes with the attributes that have the
+      // same names from the child. If the corresponding attribute is not found, throw an
+      // AnalysisException.
+      // TODO: Also check the dataTypes and nullabilites of the output.
```

Comment:
Can we also explain why we need to replace output attributes in the code comment?